parsons/google/google_cloud_storage.py

```diff
@@ -453,27 +450,33 @@ def copy_bucket_to_gcs(
 
         while polling:
             if latest_operation_name:
+
                 operation = client.get_operation({"name": latest_operation_name})
 
                 if not operation.done:
                     logger.debug("Operation still running...")
 
                 else:
-                    if int(operation.error.code) not in StatusCode["OK"].value:
+                    operation_metadata = storage_transfer.TransferOperation.deserialize(
+                        operation.metadata.value
+                    )
+                    error_output = operation_metadata.error_breakdowns
+                    if len(error_output) != 0:
                         raise Exception(
-                            f"""{blob_storage} to GCS Transfer Job {create_result.name} failed with error: {operation.error.message}
+                            f"""{blob_storage} to GCS Transfer Job {create_result.name} failed with error: {error_output}
                             """
                         )
-                    if operation.response:
+                    else:
```

Comment:
Strictly speaking, you don't need the `else` here because the `if` clause above it will terminate execution of the method. Sometimes people like to leave the `else` in those cases to make it clearer, which is fine. (I usually do that myself, actually.)

I do think it's strange that the `return` is _not_ in the else, though. The execution in the non-error path will be to skip the if, run one line in the else, and then immediately run the return. It seems like either the log and the return should be in the else, or it shouldn't be there at all.